lib/chef/validation/ext/hash.rb

```diff
@@ -22,11 +22,16 @@ class << self
       # @return [Object, nil]
       def dig(hash, path, separator = "/")
         return nil unless !path.nil? && !path.empty?
-        return nil unless hash.respond_to?(:has_key?)
-        return hash unless hash.respond_to?(:[])
 
         key, rest = path.split(separator, 2)
-        match     = hash[key.to_s].nil? ? hash[key.to_sym] : hash[key.to_s]
+        if hash.is_a?(Hash)
+          return nil unless hash.respond_to?(:has_key?)
```

Comment:
You don't need this check or the one below it since we know that instances of Hash repond to these messages